modules/setting/mirror.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package setting
+
+import (
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+)
+
+var (
+	// Mirror settings
+	Mirror = struct {
+		Enabled         bool
+		DisableNewPull  bool
+		DisableNewPush  bool
+		DefaultInterval time.Duration
+		MinInterval     time.Duration
+	}{
+		Enabled:         true,
+		DisableNewPull:  false,
+		DisableNewPush:  false,
+		MinInterval:     10 * time.Minute,
+		DefaultInterval: 8 * time.Hour,
+	}
+)
+
+func newMirror() {
+	// Handle old configuration through `[repository]` `DISABLE_MIRRORS`
+	// - please note this was badly named and only disabled the creation of new pull mirrors
+	if Cfg.Section("repository").Key("DISABLE_MIRRORS").MustBool(false) {
+		log.Warn("Deprecated DISABLE_MIRRORS config is used, please change your config and use the options within the [mirror] section")
+		// TODO: enable on v1.17.0: log.Error("Deprecated fallback used, will be removed in v1.18.0")
+		Mirror.DisableNewPull = true
```

Comment:
Only pull mirrors?
But not push mirrors?

Reply:
@delvh yes - it just mirror the current behaviour ... (after all it's a fallback and will be removed later)